src/test/java/itest/GraphQLIT.java

```diff
@@ -588,6 +602,392 @@ void testNodesHaveIds() throws Exception {
         }
     }
 
+    @Test
+    @Order(9)
+    void testQueryForSpecificTargetsByNames() throws Exception {
+        CompletableFuture<TargetNodesQueryResponse> resp = new CompletableFuture<>();
+
+        String query =
+                String.format(
+                        "query { targetNodes(filter: { names:"
+                            + " [\"service:jmx:rmi:///jndi/rmi://cryostat-itests:9091/jmxrmi\","
+                            + " \"service:jmx:rmi:///jndi/rmi://cryostat-itests:9093/jmxrmi\"] }) {"
+                            + " name nodeType } }");
+        webClient
+                .post("/api/v2.2/graphql")
+                .sendJson(
+                        new JsonObject().put("query", query),
+                        ar -> {
+                            if (assertRequestStatus(ar, resp)) {
+                                resp.complete(
+                                        gson.fromJson(
+                                                ar.result().bodyAsString(),
+                                                TargetNodesQueryResponse.class));
+                            }
+                        });
+        TargetNodesQueryResponse actual = resp.get(REQUEST_TIMEOUT_SECONDS, TimeUnit.SECONDS);
+        List<TargetNode> targetNodes = actual.data.targetNodes;
+
+        int expectedSize = 2;
+
+        assertThat(targetNodes.size(), is(expectedSize));
+
+        TargetNode target1 = new TargetNode();
+        target1.name = "service:jmx:rmi:///jndi/rmi://cryostat-itests:9091/jmxrmi";
+        target1.nodeType = "JVM";
+        TargetNode target2 = new TargetNode();
+        target2.name = "service:jmx:rmi:///jndi/rmi://cryostat-itests:9093/jmxrmi";
+        target2.nodeType = "JVM";
+
+        assertThat(targetNodes, hasItem(target1));
+        assertThat(targetNodes, hasItem(target2));
+    }
+
+    @Test
+    @Order(10)
+    public void testQueryForFilteredActiveRecordingsByNames() throws Exception {
```

Comment:
Shouldn't this test be calling the `/api/v2.2/graphql` endpoint at some point? I don't think it is currently exercising the active recording "names" filter.

Reply:
in my thinking then we do not need to create any active recordings? just retrieve any that are present then if any that are present, based on the data I will filter by names? The created recordings with the code that I have cannot be retrieved with ``` /api/v2.2/graphql ```. its always returning 0 for active recordings available

Reply:
never mind i think I saw something useful by creating a snapshot with ``` POST /api/v2/targets/:targetId/snapshot ``` this should work

Reply:
If you do not create any active recordings, there will be none there when you query and therefore none to filter...

> The created recordings with the code that I have cannot be retrieved with /api/v2.2/graphql. its always returning 0 for active recordings available

Then something is wrong, because these things should work together. Either there is a bug or your test query setup isn't working how you intended.

> never mind i think I saw something useful by creating a snapshot with POST /api/v2/targets/:targetId/snapshot this should work

I don't think so - creating a snapshot when there are no active recordings does not do anything.

Reply:
I might need help tomorrow if by any chance you have time. I think the problem is my graphql query. I read the .md files and ```"/api/v2.2/graphql"``` should be able to retrieve the created recordings

Reply:
Sure, just ping me or send a calendar invite.

Reply:
I will ping you... and when you are free we can meet 
mean while this is what i have 
``` CompletableFuture<TargetNodesQueryResponse> resp2 = new CompletableFuture<>();
 String query =
     "query { targetNodes {"
             + "name"
             + "nodeType"
             + "labels"
             + "target {"
             + "alias"
             + "serviceUri"
             + "}"
             + "recordings {"
             + "active {"
             + "data {"
             + "name"
             + "state"
             + "metadata {"
             + "labels"
             + "}"
             + "}"
             + "aggregate {"
             + "count"
             + "}"
             + "}"
             + "}"
             + "}"
             + "}";
 webClient.post("/api/v2.2/graphql")
     .sendJson(new JsonObject().put("query", query), ar -> {
         if (assertRequestStatus(ar, resp2)) {
             resp2.complete(
                     gson.fromJson(
                             ar.result().bodyAsString(),
                             TargetNodesQueryResponse.class));
         }
     });
 
     TargetNodesQueryResponse graphqlResp = resp2.get(REQUEST_TIMEOUT_SECONDS, TimeUnit.SECONDS);``` trying to get all active recordings... i created two named Recording1 and Recording2